tests/Client/TestRepo.php

```diff
@@ -74,15 +61,34 @@ public function fetchFileIfExists(string $fileName, int $maxBytes):?string
     }
 
     /**
-     * Sets the file for which a signature fail should be triggered.
+     * Sets a nested value in a repo file.
+     *
+     * @param string $fileName
+     *   The name of the file to change.
+     * @param array $keys
+     *   The nested array keys of the item.
+     * @param mixed $newValue
+     *   The new value to set.
+     *
+     * @return void
+     */
+    public function setRepoFileNestedValue(string $fileName, array $keys = ['signed', 'extra_test_value'], $newValue = 'new value'): void
```

Comment:
Why do $keys and $newValue have defaults? That seems strange.

Reply:
So we can used the method to set a nested value when we don't actually care what the value is just that contents have changed. Right now this is the only use case for this method but it is being pull from #80 which hopefully will be in soon. In that case we do have case we want to change a specific key.